### DISCUSSION TITLE
Multipolygon union

### DISCUSSION THREAD

**Discussion by huntto** [boostorg/geometry#1125] [General]
My multipolygon consists of multiple polygons, but these polygons intersect in pairs. If I perform union operation, can I get only one polygon? The code is as follows:

```
#include <boost/geometry.hpp>
#include <boost/foreach.hpp>
#include <boost/geometry/geometries/polygon.hpp>
#include <boost/geometry/geometries/adapted/boost_tuple.hpp>

typedef boost::geometry::model::d2::point_xy<int> Point;
typedef boost::geometry::model::polygon<Point, false> Polygon;
typedef boost::geometry::model::multi_polygon<Polygon> MultiPolygon;
typedef boost::geometry::model::box<Point> Box;

int main() {

    Polygon p1{{{1, 1}, {3, 1}, {3, 2}, {1, 2}}};
    Polygon p2{{{2, 1}, {5, 1}, {5, 2}, {2, 2}}};
    Polygon p3{{{4, 1}, {7, 1}, {7, 2}, {4, 2}}};
    Polygon p4{{{6, 1}, {8, 1}, {8, 2}, {6, 2}}};

    MultiPolygon mp1{p1, p2, p3};

    MultiPolygon mp2{p4};

    MultiPolygon output;
    boost::geometry::union_(mp1, mp2, output);
    std::cout << "output size:" << output.size() << std::endl;
    return 0;
}
```

I want the `output` to contain only one polygon.

**Comment by awulkiew** [comment]
Try
```
std::string message;
boost::geometry::is_valid(mp1, message);
```
and see what happens.

**Comment by barendgehrels** [comment]
In other words, you should union the polygons one-by-one.
So p1+p2 -> p12
then p12 + p3 -> p123
then p123 + p4 -> p1234

I once wrote a more efficient implemention for this (using boxes and combining what appropriate) but ATM I cannot remember where it is in github (apart from somewhere on my harddisk)

**Reply by 0116lhc** [reply]
Hello, I have encountered this problem as well. I need to merge many polygons into one polygon, only requiring the entire outer contour after the merging. Could you please give me some hints?
